File: 조준장/level2/다리를 지나는 트럭.kt
class Solution {
    fun solution(bridge_length: Int, weight: Int, truck_weights: IntArray): Int {
        var answer = 0
        var truckWeights = truck_weights.toMutableList()
        var bridge = MutableList(bridge_length){0}
        
        while (!bridge.isEmpty()){
            answer++
            bridge.removeFirst()
            if (!truckWeights.isEmpty()){
                if (bridge.sum() + truckWeights[0] <= weight){
                    bridge.add(truckWeights.removeFirst())
                    continue
                } 
                bridge.add(0)
                
            }
            
        }
        
        
        return answer
    }
}

File: 이건희/level2/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    result = 0
    order = [0] * bridge_length
    truck = truck_weights
    sum_weight = 0
    
    while order:
        result += 1
        sum_weight -= order.pop(0)
        
        if truck:
            if sum_weight + truck[0] <= weight:
                sum_weight += truck[0]
                order.append(truck.pop(0))
            else:
                order.append(0)
                
    return result 
